cardano-cli/src/Cardano/CLI/Shelley/Run/Query.hs

```diff
@@ -567,86 +562,28 @@ runQueryKesPeriodInfo (AnyConsensusModeParams cModeParams) network nodeOpCertFil
-                 "The protocol state counter is greater than the counter in the operational certificate at: " <> nodeOpCertFile' <> "\n" <>
-                 "On disk operational certificate count: " <> show onDiskOpCertCount <> "\n" <>
-                 "Protocol state count: " <> show ptclStateCounter
-
-                OpCertKesPeriodOutsideOfCurrentInterval (nodeOpCertFile', opCertKesPeriod) intervalStart intervalEnd ->
-                  "Node operational certificate at: " <> nodeOpCertFile' <> " has an incorrectly specified KES period. " <> "\n" <>
-                  "The operational certificate's specified KES period is outside of the current KES period interval." <> "\n" <>
-                  "KES period interval start: " <> show intervalStart <> "\n" <>
-                  "KES period interval end: " <> show intervalEnd <> "\n" <>
-                  "Operational certificate's specified KES period: " <> show opCertKesPeriod
-
-          in putStrLn $ "✗ " <> failureString
-
-data KesOpCertDiagnostic = SuccessDiagnostic SuccessDiagnostic
-                         | FailureDiagnostic FailureDiagnostic
-
-
-anyFailureDiagnostic :: KesOpCertDiagnostic -> Bool
-anyFailureDiagnostic FailureDiagnostic{} = True
-anyFailureDiagnostic SuccessDiagnostic{} = False
-
-data SuccessDiagnostic
-  = OpCertCounterMoreThanOrEqualToNodeState
-  | OpCertCounterMoreThanOrEqualToZero
-  | OpCertCurrentKesPeriodWithinInterval
-  | OpCertCountersAreEqual
-
-data FailureDiagnostic
-  = OpCertCounterLessThanPtclCounter
-      Word64
-      -- ^ Operational certificate counter in the protocol state
-      Word64
-      -- ^ Operational certificate counter on disk
-      FilePath
-      -- ^ Operational certificate
-  | OpCertKesPeriodOutsideOfCurrentInterval
-      !(FilePath, Word64)
-      -- ^ Operational certificate filepath with its KES period
-      !Word64
-      -- ^ Kes Period interval start
-      !Word64
-      -- ^ Kes Period interval end
-  | OpCertCounterNegative
+          Just ptclStateCounter -> return (OpCertOnDiskCounter onDiskOpCertCount, Just $ OpCertNodeStateCounter ptclStateCounter)
+          Nothing -> return (OpCertOnDiskCounter onDiskOpCertCount, Nothing)
+
+renderOpCertIntervalInformation :: FilePath -> OpCertIntervalInformation -> String
+renderOpCertIntervalInformation _ (OpCertWithinInterval _start _end _current _stillExp) =
+  "✓ Operational certificate's kes period is within the correct KES period interval"
+renderOpCertIntervalInformation opCertFile
+  (OpCertStartingKesPeriodIsInTheFuture (OpCertStartingKesPeriod start) (OpCertEndingKesPeriod end) (CurrentKesPeriod current)) =
+   "✗ Node operational certificate at: " <> opCertFile <> " has an incorrectly specified starting KES period. " <> "\n" <>
+   "Current KES period: " <> show current <> "\n" <>
+   "Operational certificate's starting KES period: " <> show start <> "\n" <>
+   "Operational certificate's expiry KES period: " <> show end
```

Comment:
Perhaps we should indent this lines so the output looks like this to aid readability?

```
✗ Node operational certificate at ...
   Current KES period: ...
   Operational certificate's starting KES period: ...
   Operational certificate's expiry KES period: ...
```

Similarly below.